packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -0,0 +1,368 @@
+  /**
+   * An ISO 8601 string representation of the maximum date time.
+   */
+  maxDateTime: PropTypes.string,
+  /**
+   * An ISO 8601 string representation of the minimum date time.
+   */
+  minDateTime: PropTypes.string,
+  /**
+   * Name of the date input. The name should be unique.
+   */
+  name: PropTypes.string.isRequired,
+  /**
+   * A callback function to execute when a valid date is selected or entered.
+   * The first parameter is the event. The second parameter is the changed input value.
+   */
+  onChange: PropTypes.func,
+  /**
+   * A callback function to execute when a change is made in the date or time input.
+   * The first parameter is the event. The second parameter is the changed input value.
+   */
+  onChangeRaw: PropTypes.func,
+  /**
+   * A callback function to let the containing component (e.g. modal) to regain focus.
+   */
+  releaseFocus: PropTypes.func,
+  /**
+   * A callback function to request focus from the containing component (e.g. modal).
+   */
+  requestFocus: PropTypes.func,
+  /**
+   * Custom input attributes to apply to the time input. Use the name prop to set the name for the time input.
+   * Do not set the name in inputAttribute as it will be ignored.
+   */
+  timeInputAttributes: PropTypes.object,
+  /**
+   * An ISO 8601 string representation of the initial value to show in the date and time inputs.
+   */
+  value: PropTypes.string,
+};
+
+const defaultProps = {
+  dateInputAttributes: undefined,
+  excludeDates: undefined,
+  filterDate: undefined,
+  includeDates: undefined,
+  maxDateTime: undefined,
+  minDateTime: undefined,
+  name: undefined,
+  onChange: undefined,
+  onChangeRaw: undefined,
+  releaseFocus: undefined,
+  requestFocus: undefined,
+  timeInputAttributes: undefined,
+  value: undefined,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+const keyCodes = {
+  ARROWDOWN: 40,
+};
+
+class DateTimePicker extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      dateTime: DateTimeUtils.createSafeDate(props.value),
+      isAmbiguousTime: false,
+      isTimeClarificationOpen: false,
+    };
+
+    // The dateValue and timeValue variables represent the actual value in the date input and time input respectively.
+    // They are used to keep track of the currently entered value to determine whether or not the entry is valid.
+    // Unlike dateValue and timeValue, this.state.dateTime is the internal moment object representing both the date and time as one entity
+    // It is used for date/time minuipulation and used to calculate the missing/ambiguous hour.
+    // The dateValue and timeValue are tracked outside of the react state to limit the number of renderings that occur.
+    this.dateValue = '';
+    this.timeValue = '';
```

Comment:
maybe put these in default props?

Reply:
These are just properties on the component and not props so putting them in defaultProps would actually throw an error from PropsTable.